### ISSUE TITLE
`PIE796` false-positives on enum member cast in pyi files

### ISSUE THREAD

**Issue by Avasam** [astral-sh/ruff#15132]
Found this whilst applying various Ruff groups on typeshed out of curiosity.

According to the updated typing spec for typing enums (see [Mypy: Change to Enum Membership Semantics](https://mypy-lang.blogspot.com/2024/12/mypy-114-released.html) and https://typing.readthedocs.io/en/latest/spec/enums.html#defining-members, this is the recommended way to type an enum in stubs where the type is known but the value isn't:
```py
class Key(enum.Enum):
    alt = cast(KeyCode, ...)
    alt_l = cast(KeyCode, ...)
    alt_r = cast(KeyCode, ...)
    alt_gr = cast(KeyCode, ...)
    # ...
```
This will trigger `PIE796`

However I think that a fix could be simpler than that: This rule probably shouldn't apply to stubs as it's out of the control of a stub's author (ref https://github.com/astral-sh/ruff/issues/14535 )

ruff 0.8.4
command: ruff check --select=PIE --isolated
